File: src/random.rs
use sha2::{Digest, Sha256};

fn hash_seed(board_id: &str, username: &str, timestamp: u64) -> u32 {
    let mut hasher = Sha256::new();
    hasher.update(board_id);
    hasher.update(username);
    hasher.update(timestamp.to_string());
    let result = hasher.finalize();
    u32::from_le_bytes(result[0..4].try_into().unwrap()) // First 4 bytes as u32
}

pub fn rnd_range(board_id: &str, username: &str, timestamp: u64, min: u32, max: u32) -> u32 {
    let seed = hash_seed(board_id, username, timestamp);
    (seed % (max - min)) + min
}

File: website/src/lib/utils/random.ts
import { Sha256 } from '@aws-crypto/sha256-browser';

export const hashSeed = async (
	boardId: string,
	username: string,
	timestamp: string
): Promise<number> => {
	const hasher = new Sha256();
	hasher.update(boardId);
	hasher.update(username);
	hasher.update(timestamp);
	const hash = await hasher.digest();
	const dataView = new DataView(hash.buffer);
	return dataView.getUint32(0, true); // Read the first 4 bytes as a little-endian u32
};

export const rndRange = async (
	boardId: string,
	username: string,
	timestamp: string,
	min: number,
	max: number
): Promise<number> => {
	const seed = await hashSeed(boardId, username, timestamp);
	return (seed % (max - min)) + min;
};
